lib/cache/strategies/labshare.cache.ts

```diff
@@ -1,12 +1,37 @@
 import {IStorage} from '../storages/IStorage';
-import {AbstractBaseStrategy} from './abstract.base.strategy';
 import {IOptions, IExpiringCacheItem} from '../types';
 import {CacheConstants} from '../constants/index';
 import * as _ from 'lodash';
+import {RedisStorage} from '../storages/redis.storage';
+import {MemoryStorage} from '../storages/memory.storage';
 
-export class LabShareCache extends AbstractBaseStrategy {
-  constructor(storage: IStorage) {
-    super(storage);
+export class LabShareCache {
+  private storage: IStorage;
+  constructor(public cacheConfig: any) {
+    this.storage = this.createCacheClient();
+  }
+  private createCacheClient(): any {
+    let provider;
+    if (
+      _.get(
+        this.cacheConfig,
```

Comment:
As I suggested above let's have cache config as a typed object. Then you won't need to use _.get() method, which is weakly typed.